internal/repository/root_test.go

```diff
@@ -43,6 +42,106 @@ func TestInitializeRoot(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestAddRootKey(t *testing.T) {
+	r, keyBytes := createTestRepositoryWithRoot(t, "")
+
+	key, err := tuf.LoadKeyFromBytes(keyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sv, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(keyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var newRootKey *sslibsignerverifier.SSLibKey
+
+	newRootKey, err = tuf.LoadKeyFromBytes(targetsKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = r.AddRootKey(context.Background(), keyBytes, targetsKeyBytes, false)
+	assert.Nil(t, err)
+
+	state, err := policy.LoadCurrentState(context.Background(), r.r)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rootMetadata, err := state.GetRootMetadata()
+	assert.Nil(t, err)
+	assert.Equal(t, 2, rootMetadata.Version)
+	assert.Equal(t, []string{key.KeyID, newRootKey.KeyID}, rootMetadata.Roles[policy.RootRoleName].KeyIDs)
+	assert.Equal(t, key.KeyID, state.RootEnvelope.Signatures[0].KeyID)
+
+	err = dsse.VerifyEnvelope(context.Background(), state.RootEnvelope, []sslibdsse.Verifier{sv}, 1)
+	assert.Nil(t, err)
+}
+
+func TestRemoveRootKey(t *testing.T) {
+	r, keyBytes := createTestRepositoryWithRoot(t, "")
+
+	rootKey, err := tuf.LoadKeyFromBytes(keyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sv, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(keyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = r.AddRootKey(context.Background(), keyBytes, keyBytes, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	newRootKey, err := tuf.LoadKeyFromBytes(targetsKeyBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = r.AddRootKey(context.Background(), keyBytes, targetsKeyBytes, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	state, err := policy.LoadCurrentState(context.Background(), r.r)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rootMetadata, err := state.GetRootMetadata()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, 3, rootMetadata.Version)
+	assert.Equal(t, rootKey.KeyID, rootMetadata.Roles[policy.RootRoleName].KeyIDs[0])
```

Comment:
This check can be removed given the `Contains` checks below.

Reply:
Removed